src/graphql/node/control.rs

```diff
@@ -42,4 +60,1065 @@ impl NodeControlMutation {
             )
         }
     }
+
+    #[graphql(guard = "RoleGuard::new(Role::SystemAdministrator)
+        .or(RoleGuard::new(Role::SecurityAdministrator))")]
+    async fn apply_node(&self, ctx: &Context<'_>, id: ID) -> async_graphql::Result<ID> {
+        let i = id.as_str().parse::<u32>().map_err(|_| "invalid ID")?;
+
+        let node: Node = {
+            let store = crate::graphql::get_store(ctx).await?;
+            let node_map = store.node_map();
+            node_map
+                .get_by_id(i)?
+                .ok_or_else(|| async_graphql::Error::new(format!("Node with ID {i} not found",)))?
+        };
+        if node.setting_draft.is_none() {
+            return Err("Cannot apply when `setting_draft` is None".into());
```

Comment:
Both `setting_draft` and `None` are code-centric. I think error messages should be plain texts, which are contextually descriptive.